eth/sync.go

```diff
@@ -162,7 +162,7 @@ func (pm *ProtocolManager) synchronise(peer *peer) {
 		return
 	}
 	// Make sure the peer's TD is higher than our own. If not drop.
-	td := pm.blockchain.GetTd(pm.blockchain.CurrentBlock().Hash())
+	td := pm.blockchain.GetTd(pm.blockchain.CurrentBlock().Hash(), pm.blockchain.CurrentBlock().NumberU64())
```

Comment:
This is a race condition. Current block might change between two invocations, causing `td` to be nil. Please query the current block once and fetch the hash/number from that.